R/expectation.r

```diff
@@ -11,20 +11,73 @@
 #' @keywords internal
 #' @export
 expectation <- function(passed, failure_msg, success_msg = "unknown", srcref = NULL) {
-  structure(
+  new_expectation(passed = passed, failure_msg = failure_msg,
+                  success_msg = success_msg, srcref = srcref)
+}
+
+new_expectation <- function(failure_msg, success_msg, srcref, ...,
+                            passed = FALSE, error = FALSE, skipped = FALSE) {
+  if (passed) {
```

Comment:
Sure. I think I just introduced some merge conflicts, but if you squash this, I'll fix.